### PR TITLE
chore: add linting

### PR DESCRIPTION
**Pull request by dabeeeenster**
- Adds pre-commit config
- Runs pre-commit checks as a Github Action

### CODE REVIEW

**Comment by riceyrice** [comment]
This brings in competing `prettier` config between the top-level Forge CLI folder and the Jira app folder it contains. The inner folder was intended to be self-contained with the outer folder as a convenience to avoid having to install Forge CLI globally. Notably the outer and inner folder use different versions of Node.js.

**Comment by dabeeeenster** [comment]
Ah - I missed the linting in the app folder. What do you think @matthewelwell ? We're trying to standardise on pre-commit I feel?

**Comment by matthewelwell** [comment]
Can anyone dumb this down for me so I don't have to understand the structure of the project? What's the issue and what are the options? @dabeeeenster @riceyrice

**Comment by riceyrice** [comment]
> Can anyone dumb this down for me so I don't have to understand the structure of the project? What's the issue and what are the options? @dabeeeenster @riceyrice

@matthewelwell the root of the repo contains/installs a Node 20 app (the Forge CLI). The `flagsmith-jira-app` folder contains/builds a Node 16 app (the Jira app). Both create a `node_modules` directory with different dependencies.

It's OK to install linting/formatting tools in the root if you need to make this work with pre-commit - or maybe pre-commit can be made to work with the linting/formatting tools already installed in the `flagsmith-jira-app` folder. If you do put the tools/config in the root folder, the existing tools/config in the app folder should be removed to avoid conflicts.

**Comment by dabeeeenster** [comment]
OK that should be in line with what we want now

File: flagsmith-jira-app/.prettierrc
Comment (matthewelwell) [inline flagsmith-jira-app/.prettierrc:1]: Shouldn't this have been removed as per discussion in the main PR thread?

Comment (dabeeeenster) [reply flagsmith-jira-app/.prettierrc:1]: Its needed in both locations for both to work AFAICT

**Review by matthewelwell — APPROVED** [review]

**Comment by matthewelwell** [comment]
@dabeeeenster not sure what the status of this one is but it seems ok to me, shall we merge?

**Pull request merged**